test/test_tensorflow.py

```diff
@@ -1374,6 +1376,10 @@ def test_horovod_broadcast_grad_gpu(self):
             # Skip if compiled with CUDA but without HOROVOD_GPU_OPERATIONS.
             self.skipTest("Not compiled with HOROVOD_GPU_OPERATIONS")
 
+        # This test does not apply if using gloo controller
+        if hvd.gloo_enabled():
```

Comment:
Allgather GPU does not work on Gloo?

Reply:
Nice catch. Got a little overzealous with the Gloo test skips it seems :). Removed.